test/stdlib/Encoding_Converter_test.rb

```diff
@@ -0,0 +1,364 @@
+    )
+  end
+
+  def test_convert
+    assert_send_type(
+      "(String) -> String",
+      Encoding::Converter.new("UTF-8", "EUC-JP"), :convert, "\u{3042}",
+    )
+  end
+
+  def test_convpath
+    assert_send_type(
+      "() -> [[Encoding, Encoding]]",
+      Encoding::Converter.new("UTF-8", "EUC-JP"), :convpath
+    )
+  end
+
+  def test_destination_encoding
+    assert_send_type(
+      "() -> Encoding",
+      Encoding::Converter.new("UTF-8", "EUC-JP"), :destination_encoding
+    )
+  end
+
+  def test_finish
+    assert_send_type(
+      "() -> String",
+      Encoding::Converter.new("UTF-8", "EUC-JP"), :finish
+    )
+  end
+
+  def test_insert_output
+    assert_send_type(
+      "(String) -> nil",
+      Encoding::Converter.new("UTF-8", "EUC-JP"), :insert_output, "Foo"
+    )
+  end
+
+  def test_last_error
+    invalid_ec = Encoding::Converter.new("UTF-8", "ISO-8859-1")
+    invalid_ec.primitive_convert("\xf1abcd", "")
+    assert_send_type(
+      "() -> Encoding::InvalidByteSequenceError",
+      invalid_ec, :last_error
+    )
+
+    undefined_ec = Encoding::Converter.new("ISO-8859-1", "EUC-JP")
+    undefined_ec.primitive_convert("\xa0", "")
+    assert_send_type(
+      "() -> Encoding::UndefinedConversionError",
+      undefined_ec, :last_error
+    )
+  end
+
+  def test_primitive_convert
+    assert_send_type(
+      "(String, String) -> :finished",
+      Encoding::Converter.new("UTF-8", "UTF-16BE"), :primitive_convert,
+      "Foo", ""
+    )
+    assert_send_type(
+      "(String, String, nil, nil) -> :finished",
+      Encoding::Converter.new("UTF-8", "UTF-16BE"), :primitive_convert,
+      "Foo", "", nil, nil
+    )
+    assert_send_type(
+      "(String, String, nil, Integer) -> :destination_buffer_full",
+      Encoding::Converter.new("UTF-8", "UTF-16BE"), :primitive_convert,
+      "Foo", "", nil, 1
+    )
+    assert_send_type(
+      "(String, String, nil, Integer) -> :invalid_byte_sequence",
+      Encoding::Converter.new("EUC-JP", "Shift_JIS"), :primitive_convert,
+      "\xff", "", nil, 10
+    )
+    assert_send_type(
+      "(String, String, nil, Integer) -> :undefined_conversion",
+      Encoding::Converter.new("EUC-JP", "ISO-8859-1"), :primitive_convert,
+      "\xa4\xa2", "", nil, 10
+    )
+    assert_send_type(
+      "(String, String, nil, Integer) -> :incomplete_input",
+      Encoding::Converter.new("EUC-JP", "ISO-8859-1"), :primitive_convert,
+      "\xa4", "", nil, 10
+    )
+    assert_send_type(
+      "(String, String, nil, Integer, Integer) -> :source_buffer_empty",
+      Encoding::Converter.new("EUC-JP", "ISO-8859-1"), :primitive_convert,
+      "\xa4", "", nil, 10, Encoding::Converter::PARTIAL_INPUT
+    )
+    assert_send_type(
+      "(String, String, nil, Integer, partial_input: true) -> :source_buffer_empty",
+      Encoding::Converter.new("EUC-JP", "ISO-8859-1"), :primitive_convert,
+      "\xa4", "", nil, 10, partial_input: true
+    )
+
+    # TODO: Add after_output test case
```

Comment:
I don't know what cases return `:after_output`, so I haven't implemented that test.